sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreClient.java

```diff
@@ -132,24 +131,24 @@ private RxDocumentServiceResponse completeResponse(
             throw new InternalServerErrorException(RMResources.InvalidBackendResponse);
         }
 
-        Map<String, String> headers = new HashMap<>(storeResponse.getResponseHeaderNames().length);
+        HttpHeaders httpHeaders = new HttpHeaders();
```

Comment:
potential perf impact. previously we used to pass the capacity to the constructor of the hashmap upfront reserving the required capacity whereas now the underlying map can grow multiple times.